### PR TITLE
[API] Remove old deprecated methods.

### PR DESCRIPTION
**Pull request by garrettmoon**
I'll be going through and marking when each method was deprecated. Please hold off on reviewing until I say I'm ready.

### CODE REVIEW

File: AsyncDisplayKit/ASDisplayNode+Deprecated.h [hunk 1]
```diff
@@ -104,8 +104,4 @@
  */
 @property (nonatomic, assign) BOOL usesImplicitHierarchyManagement ASDISPLAYNODE_DEPRECATED;
 
-- (void)reclaimMemory ASDISPLAYNODE_DEPRECATED;
-- (void)recursivelyReclaimMemory ASDISPLAYNODE_DEPRECATED;
-@property (nonatomic, assign) BOOL placeholderFadesOut ASDISPLAYNODE_DEPRECATED;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASDisplayNode+Deprecated.h]: All three of the above were deprecated in 1.2

File: AsyncDisplayKit/Details/ASImageProtocols.h [hunk 2]
```diff
@@ -213,29 +213,4 @@ withDownloadIdentifier:(id)downloadIdentifier;
 
 @end
 
-@protocol ASImageDownloaderProtocolDeprecated <ASImageDownloaderProtocol>
-
-@optional
-/**
- @deprecated This method is deprecated @see downloadImageWithURL:callbackQueue:downloadProgress:completion: instead
- */
-- (nullable id)downloadImageWithURL:(NSURL *)URL
```
Comment (garrettmoon) [inline AsyncDisplayKit/Details/ASImageProtocols.h]: Deprecated 1.9.7

File: AsyncDisplayKit/Details/ASImageProtocols.h [hunk 3]
```diff
@@ -213,29 +213,4 @@ withDownloadIdentifier:(id)downloadIdentifier;
...
-
-@end
-
-@protocol ASImageCacheProtocolDeprecated <ASImageCacheProtocol>
-
-@optional
-/**
- @deprecated This method is deprecated @see cachedImageWithURL:callbackQueue:completion: instead
- */
-- (void)fetchCachedImageWithURL:(nullable NSURL *)URL
```
Comment (garrettmoon) [inline AsyncDisplayKit/Details/ASImageProtocols.h]: Deprecated 1.9.7

File: AsyncDisplayKit/ASTableView.h [hunk 4]
```diff
@@ -347,26 +347,6 @@ NS_ASSUME_NONNULL_BEGIN
...
 
-/**
- * Indicator to lock the data source for data fetching in async mode.
- * We should not update the data source until the data source has been unlocked. Otherwise, it will incur data inconsistency or exception
- * due to the data access in async mode.
- *
- * @param tableView The sender.
- * @deprecated The data source is always accessed on the main thread, and this method will not be called.
- */
-- (void)tableViewLockDataSource:(ASTableView *)tableView ASDISPLAYNODE_DEPRECATED;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASTableView.h]: Deprecated 1.9.81

File: AsyncDisplayKit/ASTableView.h [hunk 5]
```diff
@@ -347,26 +347,6 @@ NS_ASSUME_NONNULL_BEGIN
...
-
-/**
- * Indicator to unlock the data source for data fetching in asyn mode.
- * We should not update the data source until the data source has been unlocked. Otherwise, it will incur data inconsistency or exception
- * due to the data access in async mode.
- *
- * @param tableView The sender.
- * @deprecated The data source is always accessed on the main thread, and this method will not be called.
- */
-- (void)tableViewUnlockDataSource:(ASTableView *)tableView ASDISPLAYNODE_DEPRECATED;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASTableView.h]: Deprecated 1.9.81

File: AsyncDisplayKit/ASTableView.h [hunk 6]
```diff
@@ -450,42 +430,9 @@ NS_ASSUME_NONNULL_BEGIN
...
- * Informs the delegate that the table view did remove the node which was previously
- * at the given index path from the view hierarchy.
- *
- * @warning AsyncDisplayKit processes table view edits asynchronously. The index path
- *   passed into this method may not correspond to the same item in your data source
- *   if your data source has been updated since the last edit was processed.
- *
- * This method is deprecated. Use @c tableView:didEndDisplayingNode:forRowAtIndexPath: instead.
- */
-- (void)tableView:(ASTableView *)tableView didEndDisplayingNodeForRowAtIndexPath:(NSIndexPath *)indexPath ASDISPLAYNODE_DEPRECATED;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASTableView.h]: Deprecated 1.9.5

File: AsyncDisplayKit/ASTableView.h [hunk 7]
```diff
@@ -450,42 +430,9 @@ NS_ASSUME_NONNULL_BEGIN
...
- * @param tableView The sender.
- * @param indexPath The index path of the row that will be displayed.
- *
- * @warning AsyncDisplayKit processes table view edits asynchronously. The index path
- *   passed into this method may not correspond to the same item in your data source
- *   if your data source has been updated since the last edit was processed.
- *
- * This method is deprecated. Use @c tableView:willDisplayNode:forRowAtIndexPath: instead.
- */
-- (void)tableView:(ASTableView *)tableView willDisplayNodeForRowAtIndexPath:(NSIndexPath *)indexPath ASDISPLAYNODE_DEPRECATED;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASTableView.h]: Deprecated 2.0

File: AsyncDisplayKit/ASTableView.h [hunk 8]
```diff
@@ -450,42 +430,9 @@ NS_ASSUME_NONNULL_BEGIN
...
-
 @end
 
 @protocol ASTableViewDelegate <ASTableDelegate>
 @end
 
-@interface ASTableView (Deprecated)
-
-- (instancetype)initWithFrame:(CGRect)frame style:(UITableViewStyle)style asyncDataFetching:(BOOL)asyncDataFetchingEnabled ASDISPLAYNODE_DEPRECATED;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASTableView.h]: Deprecated 1.9.81

File: AsyncDisplayKit/ASVideoNode.h [hunk 9]
```diff
@@ -142,11 +142,6 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)videoNodeDidRecoverFromStall:(ASVideoNode *)videoNode;
 
-// Below are deprecated methods.  To be removed in ASDK 2.0 release
-- (void)videoPlaybackDidFinish:(ASVideoNode *)videoNode __deprecated;
-- (void)videoNodeWasTapped:(ASVideoNode *)videoNode __deprecated;
-- (void)videoNode:(ASVideoNode *)videoNode didPlayToSecond:(NSTimeInterval)second __deprecated;
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASVideoNode.h]: All 3 deprecated 1.9.8

**Comment by maicki** [comment]
Overall LGTM

I'm not pretty sure about to remove the methods listed below as they were just recently deprecated.

- `- (void)collectionView:(ASCollectionView *)collectionView willDisplayNodeForItemAtIndexPath:(NSIndexPath *)indexPath ASDISPLAYNODE_DEPRECATED;
`
- `- (void)tableView:(ASTableView *)tableView willDisplayNodeForRowAtIndexPath:(NSIndexPath *)indexPath ASDISPLAYNODE_DEPRECATED;`

Rely on thoughts from @levi @Adlai-Holler @appleguy on that.

**Comment by Adlai-Holler** [comment]
I would explicitly keep:
- The data source lock/unlock. That way if they have the warning about implementing deprecated methods enabled, they'll know it's gone.
- The `insetForSectionAtIndex` method for the same reason.
- `willDisplayNodeForItemAtIndexPath` because it was only deprecated recently.

Everything else I'd be willing to toss. I don't have strong feelings about any of them.

**Comment by garrettmoon** [comment]
@Adlai-Holler I think I agree about the lock / unlock and inset methods – there's no cost to leaving them in since we don't even call them any more.

**Comment by Adlai-Holler** [comment]
Travis failure is spurious: `Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7fa1ce0dfdb0>`

**Comment by maicki** [comment]
- Agree with @Adlai-Holler for the lock / unlock and inset methods
- As I mentioned above: we should not remove `willDisplayNodeForItemAtIndexPath` yet.

**Comment by garrettmoon** [comment]
Looks like there's consensus on locking, inset and willDisplayNodeForItemAtIndexPath, I'll update the PR.

File: AsyncDisplayKit/Details/ASBasicImageDownloader.mm [hunk 10]
```diff
@@ -229,9 +227,9 @@ - (instancetype)_init
 #pragma mark ASImageDownloaderProtocol.
 
 - (id)downloadImageWithURL:(NSURL *)URL
-             callbackQueue:(dispatch_queue_t)callbackQueue
-     downloadProgressBlock:(void (^)(CGFloat))downloadProgressBlock
-                completion:(void (^)(CGImageRef, NSError *))completion
+                      callbackQueue:(dispatch_queue_t)callbackQueue
+                   downloadProgress:(nullable ASImageDownloaderProgress)downloadProgress
+                         completion:(ASImageDownloaderCompletion)completion
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/Details/ASBasicImageDownloader.mm]: Line this indentation up

**Pull request merged**